components/ContentPicker/index.js

```diff
@@ -54,64 +69,49 @@ const ContentPicker = ({
 		}
 	}
 
-	function handleSelect(item) {
-		const newContent = [...content];
-
-		newContent.unshift({
-			id: item.id,
-			type: item.subtype,
-		});
-
-		onPickChange(newContent);
-
-		setContent(newContent);
-	}
-
-	function handleItemDelete(item, sortIndex) {
-		const newContent = [...content];
-
-		newContent.splice(sortIndex, 1);
-
-		onPickChange(newContent);
-
-		setContent(newContent);
-	}
-
-	/**
-	 * Unfortunately, we had to use !important because on PickedItem we couldn't @emotion/styled css
-	 * as it was breaking sortability from react-sortable-hoc
-	 */
-	const StyleWrapper = styled('div')`
-		& .block-editor-link-control__search-item {
-			border: none !important;
-			cursor: default;
-
-			&:hover {
-				background: transparent;
-			}
+	// Run onPickChange callback when content changes.
+	useEffect(() => {
+		onPickChange(content);
+	}, [content, onPickChange]);
+
+	const handleSelect = (item) => {
+		setContent((previousContent) => [
+			{
+				id: item.id,
+				type: item.subtype,
+			},
+			...previousContent,
+		]);
+	};
+
+	const onDeleteItem = (deletedItem) => {
```

Comment:
Removed the `sortIndex` argument because it's not needed to filter the deleted item out of the content. All content types/modes (posts, terms, users) will have an `id` to use for filtering.